hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndexUtils.java

```diff
@@ -149,6 +150,9 @@ public static <R> HoodieRecord<R> tagAsNewRecordIfNeeded(HoodieRecord<R> record,
       // currentLocation 2 times and it will fail the second time. So creating a new in memory
       // copy of the hoodie record.
       HoodieRecord<R> newRecord = record.newInstance();
+      if (record instanceof HoodieEmptyRecord) {
+        newRecord = record.newInstance(record.getKey(), record.getOperation());
+      }
```

Comment:
Can this be before L152 `HoodieRecord<R> newRecord = record.newInstance();`?

Reply:
Changed to use a if-else statement to initialize the newRecord.